### PR TITLE
fix: fix misc side-nav issues

### PR DESCRIPTION
**Pull request by MondalSayantan**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-ce/components/CESideNav/CESideNavV2.tsx [hunk 1]
```diff
@@ -36,7 +35,7 @@ const CCMSideNavLinks = (mode: NAV_MODE): React.ReactElement => {
 
   return (
     <SideNav.Main disableScopeSelector>
-      <SideNav.Section className={css.section}>
+      <SideNav.Section title>
```
Comment (Sagar-Rajput7) [inline src/modules/75-ce/components/CESideNav/CESideNavV2.tsx:38]: why `title` passed here, I see we dont have a title string here

Comment (MondalSayantan) [reply src/modules/75-ce/components/CESideNav/CESideNavV2.tsx:38]: It's just a Boolean value

Comment (MondalSayantan) [reply src/modules/75-ce/components/CESideNav/CESideNavV2.tsx]: changed the name of the variable

File: src/modules/10-common/navigation/SideNavV2/SideNavSection/SideNavSection.tsx [hunk 2]
```diff
@@ -1,27 +1,18 @@
 import React from 'react'
 import cx from 'classnames'
-import { Layout, Text } from '@harness/uicore'
-import { FontVariation, Color } from '@harness/design-system'
+import { Layout } from '@harness/uicore'
 import css from '../SideNavV2.module.scss'
 
 export interface SideNavSectionComponentProps {
-  title?: string
+  removeBottomBorder?: boolean
```
Comment (avinashmadhwani02) [inline src/modules/10-common/navigation/SideNavV2/SideNavSection/SideNavSection.tsx:7]: why do we need this?

Comment (avinashmadhwani02) [reply src/modules/10-common/navigation/SideNavV2/SideNavSection/SideNavSection.tsx:7]: I can see it's used in CE. 

If we don't need border, let's not use SideNav.Section instead

File: src/modules/10-common/navigation/SideNavV2/SideNavV2.tsx [hunk 3]
```diff
@@ -184,6 +188,28 @@ interface SettingsLinkProps {
   mode: NAV_MODE
 }
 
+const SettingsTitle = (props: { title: keyof StringsMap }): JSX.Element => {
+  const { title } = props
+  const { getString } = useStrings()
+
+  return (
+    <Container>
```
Comment (avinashmadhwani02) [inline src/modules/10-common/navigation/SideNavV2/SideNavV2.tsx:196]: do we need this container?

File: src/modules/75-ce/components/CESideNav/CESideNavV2.tsx [hunk 4]
```diff
@@ -176,7 +177,8 @@ const CCMSideNavLinks = (mode: NAV_MODE): React.ReactElement => {
           />
         </SideNav.Scope>
       </SideNav.Section>
-      <SideNav.Section title={getString('ce.sideNav.sectionTitles.costGovernance')}>
+      <SideNav.Section removeBottomBorder>
+        <SideNav.Title title="ce.sideNav.sectionTitles.costGovernance" />
```
Comment (avinashmadhwani02) [inline src/modules/75-ce/components/CESideNav/CESideNavV2.tsx:181]: component name is already Title we can rename props to label?

File: src/modules/10-common/navigation/SideNavV2/SideNavSection/SideNavSection.tsx (hunk 2 above)
Comment (MondalSayantan) [reply src/modules/10-common/navigation/SideNavV2/SideNavSection/SideNavSection.tsx:7]: SideNavSection adds a bottom border. I created a boolean to remove that conditionally

File: src/modules/10-common/navigation/SideNavV2/SideNavV2.tsx (hunk 3 above)
Comment (MondalSayantan) [reply src/modules/10-common/navigation/SideNavV2/SideNavV2.tsx:196]: There was a padding issue in accordion which is being solved when I add the container

File: src/modules/75-ce/components/CESideNav/CESideNavV2.tsx (hunk 4 above)
Comment (MondalSayantan) [reply src/modules/75-ce/components/CESideNav/CESideNavV2.tsx]: done

File: src/modules/10-common/navigation/SideNavV2/SideNavV2.tsx (hunk 3 above)
Comment (MondalSayantan) [reply src/modules/10-common/navigation/SideNavV2/SideNavV2.tsx]: Removed

File: src/modules/10-common/navigation/SideNavV2/SideNavSection/SideNavSection.tsx (hunk 2 above)
Comment (MondalSayantan) [reply src/modules/10-common/navigation/SideNavV2/SideNavSection/SideNavSection.tsx]: Removed

**Comment by MondalSayantan** [comment]
Jest tests are failing due to a change in uicore https://github.com/harness/uicore/pull/1142. Failing tests are fixed in develop https://github.com/harness/harness-core-ui/pull/17925

**Review by Sagar-Rajput7 — APPROVED** [review]

**Pull request merged**